superset-frontend/src/SqlLab/components/ResultSet.jsx

```diff
@@ -238,7 +238,7 @@ export default class ResultSet extends React.PureComponent {
               <ExploreCtasResultsButton
                 table={tmpTable}
                 schema={tmpSchema}
-                dbId={query.dbId}
+                dbId={this.props.database.explore_database_visualization_id}
```

Comment:
I was thinking about doing that in the api endpoint, however this is even cleaner approach